### PR TITLE
Feedback page to see Manual grades

### LINKED ISSUE #278: Students see manual grading feedback
**Is your feature request related to a problem? Please describe.**
When manual scoring plugin is set, the students cannot see the manual scoring feedback

**Describe the solution you'd like**
When a student enter to a course, right below the `my statistics` button, we can add the access to the page where the students see their feedback. Another option might be when the students open a task that is closed, they may see their feedback.

**Additional context**
We have to wait until the PR related to the manual scoring new feature is merged.

### PR DESCRIPTION
**Pull request by Heldeg**
# Description

This PR allows the students to see the professor's feedback. For that, it adds:
- A new link on `course_menu`
- A new page where is the feedback listed 
- A new page where the feedback is shown
- A new functionality for the manual scoring. The comments area is now a codemirror for rst and also it has a tab to preview the comment

Fixes #278

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

- [ ] With a previously feedbacks for a student, see if it appear on the list and if it's possible enter to feedback and the page distribution is ok
- [ ] Write a comment on rst format and do click on preview tab. If it is empty, it shows a message.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: inginious/frontend/plugins/manual_scoring/pages/templates/feedback.html [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+$# more information about the licensing of this file.
+$#
+
+$var title: $:course.get_name(user_manager.session_language()) - Feedback
+$var Column: $:template_helper.call('course_menu',course=course,template_helper=template_helper)
+
+$def NavbarF():
+    <ol class="nav navbar-nav nav-breadcrumb">
+        <li><a href="$get_homepath()/course/$course.get_id()">$course.get_name(user_manager.session_language())</a></li>
+        <li><a href="$get_homepath()/feedback_list/$course.get_id()">$:_("Feedback List")<span class="sr-only"></span></a></li>
```
Comment (crdgonzalezca) [inline inginious/frontend/plugins/manual_scoring/pages/templates/feedback.html:14]: add a fa icon to this

File: utils/minify_plugins/minify_plugins.js [hunk 2]
```diff
@@ -265,6 +265,32 @@ function minifyManualScoringPlugin() {
     minifyCssFiles(cssFiles, cssFilesPath, "manual_scoring");
 }
 
+function minifyManualScoringPlugin() {
```
Comment (crdgonzalezca) [inline utils/minify_plugins/minify_plugins.js:253]: The function is repeated.

File: inginious/frontend/plugins/manual_scoring/pages/templates/feedback.html [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+                                <li>$line</li>
+                            </ul>
+                        </td>
+                    </tr>
+
+                </tbody>
+    </table>
+        </div>
+    <div>
+        <h3><strong>$:_("Manual grade: ")<strong id="output">$data['grade']</strong></strong></h3>
```
Comment (crdgonzalezca) [inline inginious/frontend/plugins/manual_scoring/pages/templates/feedback.html]: Move the manual scoring before the rubric

File: inginious/frontend/plugins/manual_scoring/static/js/manual_scoring_main.js [hunk 4]
```diff
@@ -29,35 +27,74 @@ function save(rubric) {
...
+}
 
-function addToggleBehaviorToProblemDescription() {
-    $(`#${TASK_NAME}`).click(function () {
-        $(`#${TASK_DESCRIPTION_TEXT_ID}`).collapse("toggle");
+function previewCode() {
+    const contentDanger = "Something went wrong";
+    const txtComment = document.getElementById(COMMENTS_TEXT_AREA_ID);
+    $(`#${PREVIEW_TAB_ID}`).click(function () {
+        $.ajax("/api/preview_content", {
```
Comment (crdgonzalezca) [inline inginious/frontend/plugins/manual_scoring/static/js/manual_scoring_main.js]: Why are you using this API, I don't actually understand the objective of the API, why don't you just send the text formatted? is it not possible or something?

File: inginious/frontend/plugins/manual_scoring/static/js/manual_scoring_main.js [hunk 5]
```diff
@@ -29,35 +27,74 @@ function save(rubric) {
...
 }
 
-function addSaveFunctionToSaveButton(rubric) {
-    $(`#${SAVE_BUTTON_ID}`).click(function () {
-        save(rubric);
+function sendManualScoringAnalytics() {
+    $.post('/api/analytics/', {
+        service: {
+            key: "manual_scoring_review",
+            name: "Manual Scoring - review"
```
Comment (crdgonzalezca) [inline inginious/frontend/plugins/manual_scoring/static/js/manual_scoring_main.js]: review? Is this the admin page? probably something like `manual scoring creation` or something like that

File: inginious/frontend/plugins/manual_scoring/static/js/rubric.js [hunk 6]
```diff
@@ -139,4 +140,59 @@ class MatrixField {
...
+    }
+
+    pickColor(score) {
+        const defaultText = "No grade";
+        const defaultColor = "#002a95"
+        if (score === defaultText) {
+            return defaultColor;
+        }
+        let color = "";
+        if (score < 0.5) {
```
Comment (crdgonzalezca) [inline inginious/frontend/plugins/manual_scoring/static/js/rubric.js]: Probably using a dictionary to map the score to a color? or using two arrays and just ask whether the score is within a range or something like that as there are many `if else` statements.

File: inginious/frontend/plugins/manual_scoring/pages/api/preview_content.py [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    return comment.parse()
+
+
+def _check_string(content):
+    """ Check if the content is a string """
+    if not isinstance(content, str):
+        raise APIError(400, "The content isn't a string")
+
+
+class PreviewContent(UserApi):
```
Comment (crdgonzalezca) [inline inginious/frontend/plugins/manual_scoring/pages/api/preview_content.py:28]: As bellow, I do not understand why this API is created as I see that it only transforms rst text, isn't it possible to just send it directly to the template when this is rendered?

**Review by crdgonzalezca — CHANGES_REQUESTED** [review]

File: inginious/frontend/plugins/manual_scoring/pages/api/preview_content.py (hunk 7 above)
Comment (crdgonzalezca) [inline inginious/frontend/plugins/manual_scoring/pages/api/preview_content.py]: Probably `RstParserAPI` is a better name

File: inginious/frontend/plugins/manual_scoring/pages/templates/course_task_list.html [hunk 8]
```diff
@@ -14,40 +14,57 @@
...
+    <div class="col-md-1 ">
+        <a type="button" class="btn btn-info" id="download_csv">
+                <i class="fa fa-table"></i> $:_("Download CSV")
+        </a>
+    </div>
+    <div class="col-md-12 py-1" id="information_area"></div>
+</div>
+
+<form method="post">
+    <div class="row justify-conte py-2">
```
Comment (crdgonzalezca) [inline inginious/frontend/plugins/manual_scoring/pages/templates/course_task_list.html]: `justify-conte`?

File: inginious/frontend/plugins/manual_scoring/pages/templates/manual_scoring.html [hunk 9]
```diff
@@ -113,19 +113,33 @@ <h4 class="my-1">$:_("Rubric")</h4>
     </table>
         </div>
     <div>
-        <strong>$:_("Current Score: ")<strong style="color: #cc0808" id="output">$data['grade']</strong></strong>
+        <h4><strong>$:_("Current Score: ")<strong id="output" style="color: #002a95">$data['grade']</strong></strong></h4>
```
Comment (crdgonzalezca) [inline inginious/frontend/plugins/manual_scoring/pages/templates/manual_scoring.html:116]: Is this color fixed? Or is it updated when fields in the rubric are selected?

File: inginious/frontend/plugins/manual_scoring/pages/templates/manual_scoring.html [hunk 10]
```diff
@@ -113,19 +113,33 @@ <h4 class="my-1">$:_("Rubric")</h4>
...
-<section class="py-2">
+<section class="py-1">
     <form method="post">
-        <h4 >$:_("Comments")</h4>
-        <div class="py-1">
-        <label for="textComments"></label>
-        <textarea class="form-control" id="textComments" rows="8">$data['comment']</textarea>
+        <h3 class="py-1">
+            $:_("Comments")<i data-toggle="tooltip" data-placement="right"
+                              title="$:_('Use rst format to write the comment')"
```
Comment (crdgonzalezca) [inline inginious/frontend/plugins/manual_scoring/pages/templates/manual_scoring.html]: I think this message is not taken in the translation or is it? please double check translations.

File: inginious/frontend/plugins/manual_scoring/__init__.py [hunk 11]
```diff
@@ -39,3 +42,12 @@ def init(plugin_manager, _, __, plugin_config):
         manual_scoring.ManualScoringPage)
 
     plugin_manager.add_hook('course_admin_menu', get_manual_scoring_link_code)
+
+    # Student pages
+
+    plugin_manager.add_page(r'/feedback_list/([a-z0-9A-Z\-_]+)', student_feedback_list.StudentFeedbackListPage)
+    plugin_manager.add_page(r'/submission_feedback/([a-z0-9A-Z\-_]+)/submission/([a-z0-9A-Z\-_]+)',
+                            feedback.FeedbackPage)
+    plugin_manager.add_page(r'/api/preview_content', PreviewContent)
```
Comment (crdgonzalezca) [inline inginious/frontend/plugins/manual_scoring/__init__.py]: Change the path, probably `/api/parse_rst`?

**Review by crdgonzalezca — CHANGES_REQUESTED** [review]

File: inginious/frontend/plugins/manual_scoring/pages/templates/manual_scoring.html (hunk 9 above)
Comment (Heldeg) [reply inginious/frontend/plugins/manual_scoring/pages/templates/manual_scoring.html:116]: It's a default color. The color changes when the rubric is modified. The first time is when the rubric is loaded, but if there is not  information about the rubric, the rubric don't change and keeps that color.

Comment (crdgonzalezca) [reply inginious/frontend/plugins/manual_scoring/pages/templates/manual_scoring.html:116]: ok, no problem.

**Review by crdgonzalezca — APPROVED** [review]

**Pull request merged**